src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java

```diff
@@ -106,84 +107,118 @@ public Vulnerability apply(VulnerableComponentView vulnerableComponentView) {
                     ? "http://web.nvd.nist.gov/view/vuln/detail?vulnId=" + vulnerableComponentView.getVulnerabilityWithRemediation().getVulnerabilityName()
                     : "");
             vulnerability.setSeverity(String.valueOf(vulnerableComponentView.getVulnerabilityWithRemediation().getSeverity()));
-            vulnerability.setScanDate(bomUpdatedValueAt);
-            return vulnerability;
-        }
-    };
+            vulnerability.setScanDate(maxBomUpdatedDate);
+            vulnerabilities.add(vulnerability);
+        });
+        return vulnerabilities;
+    }
 
-    public BlackDuckFortifyPushThread(BlackDuckFortifyMapper blackDuckFortifyMapper) {
-        this.blackDuckFortifyMapper = blackDuckFortifyMapper;
+    public BlackDuckFortifyPushThread(BlackDuckFortifyMapperGroup blackDuckFortifyMapperGroup) {
+        this.blackDuckFortifyMapperGroup = blackDuckFortifyMapperGroup;
     }
 
     @Override
     public void run() {
-        if (blackDuckFortifyMapper != null) {
-            logger.info("blackDuckFortifyMapper::" + blackDuckFortifyMapper.toString());
+        if (blackDuckFortifyMapperGroup != null) {
+            logger.info("blackDuckFortifyMapper::" + blackDuckFortifyMapperGroup.toString());
 
             ProjectVersionView projectVersionItem = null;
             List<VulnerableComponentView> vulnerableComponentViews;
+            List<HubProjectVersion> hubProjectVersions = blackDuckFortifyMapperGroup.getHubProjectVersion();
+            List<ProjectVersionView> projectVersionItems = new ArrayList<>();
+            List<Vulnerability> mergedVulnerabilities = new ArrayList<>();
+
+            // Get the last successful runtime of the job
+            Date getLastSuccessfulJobRunTime;
             try {
+                getLastSuccessfulJobRunTime = getLastSuccessfulJobRunTime(PropertyConstants.getBatchJobStatusFilePath());
+            } catch (IOException e1) {
```

Comment:
Should we handle this better than just translating it to a runtime exception? We should probably also log the actual exception here.